### ISSUE TITLE
Stop outbox cleaner process in a graceful way

### ISSUE THREAD

**Issue by yvesgoeleven** [Particular/NServiceBus.NHibernate#344]
A user reports that a warning occurs when he shutsdown an endpoint, issuing a TaskCancelledException.

Perfectly normal as we cancel the task on stop, but really unnecessary to log this

Exception occurred during stopping of feature startup task 'OutboxCleaner'. System.Threading.Tasks.TaskCanceledException: A task was canceled.     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)     at NServiceBus.Features.OutboxCleaner.<PerformCleanup>d__3.MoveNext()  --- End of stack trace from previous location where exception was thrown ---     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)     at NServiceBus.Features.FeatureStartupTaskController.<Stop>d__5.MoveNext() | AsyncTaskMethodBuilder.Start => <Stop>d__5.MoveNext => Logger.Warn

**Comment by garyamorris** [comment]
On shutdown we're currently experiencing a similar issue using latest.  
```
14:14:05 WRN] Exception occurred during stopping of feature startup task 'OutboxCleaner'.
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at NServiceBus.Features.OutboxCleaner.PerformCleanup(CancellationToken ct) in C:\BuildAgent\work\4caed983f8dafcd5\src\NServiceBus.NHibernate\Outbox\OutboxCleaner.cs:line 91
   at NServiceBus.Features.FeatureStartupTaskController.Stop(IMessageSession messageSession) in C:\BuildAgent\work\b549d46003942065\src\NServiceBus.Core\Features\FeatureStartupTaskController.cs:line 28
[14:14:05 WRN] Exception occurred during stopping of feature startup task 'OutboxCleaner'.
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at NServiceBus.Features.OutboxCleaner.PerformCleanup(CancellationToken ct) in C:\BuildAgent\work\4caed983f8dafcd5\src\NServiceBus.NHibernate\Outbox\OutboxCleaner.cs:line 91
   at NServiceBus.Features.FeatureStartupTaskController.Stop(IMessageSession messageSession) in C:\BuildAgent\work\b549d46003942065\src\NServiceBus.Core\Features\FeatureStartupTaskController.cs:line 28
```

**Comment by Tsjunne** [comment]
I would like to 1up this issue, our testers are reporting this false error regularly. 
We are running 7.1.6.

**Comment by LeMaciek** [comment]
+1 - it's annoying

**Comment by SzymonPobiega** [comment]
Fixed in #440
